classes/task/ReviewReminder.php

```diff
@@ -111,15 +44,13 @@ public function executeActions()
         $context = null;
 
         $contextDao = Application::getContextDAO();
+        $incompleteAssignments = Repo::reviewAssignment()
```

Comment:
Because you're attempting to keep the `$context` object when possible while iterating through the loop, this would be significantly faster if ordered by context ID. I see that there aren't sorting functions in the review assignment collector, but those should be pretty quick to introduce.

Reply:
In that case , we should also sort by submission id as I am trying to keep the `submission` object also if possible . I added that also .